docs/quarantine.md

```diff
@@ -0,0 +1,63 @@
+# Test Quarantine
+
+The execution of tests on CI should be as deterministic as possible, besides
+potential issues coming from infrastructure or external dependencies. When a
+test sometimes fails and sometimes passes, apparently randomly, without any
+relationship with the changes in a PR, it has very negative implications:
+* The results of the whole suite are less valuable, we can no longer trust in
+the information given by the suite execution.
+* We can't be completely sure about the state of the product, it can be ok or
```

Comment:
How is this different from the former point?

Reply:
I added the second point as a consequence of the first and more focused on the status of the product and not the status of the suite. I've rephrased it a bit to make it more clear, let me know if we should merge them.